### PR TITLE
Extended feature validations in REST tests and improved wording.

### PR DESCRIPTION
**Pull request by mosche**
Extended feature validations in REST tests and improved wording for better guidance on mistakes / misunderstandings.

### CODE REVIEW

**Review by ldematte — APPROVED** [review]
LGTM, but I'm wondering if we should also rename `createAdditionalFeatureSpecifications` to include "test" or "historical"?
And/or add the same test in the TestFeatureService assertion to the `createAdditionalFeatureSpecifications` Javadoc

**Comment by mosche** [comment]
> LGTM, but I'm wondering if we should also rename `createAdditionalFeatureSpecifications` to include "test" or "historical"?

Thanks, good point! That absolutely makes sense. I'll add that 👍

**Review by thecoop — APPROVED** [review]

File: test/framework/src/main/java/org/elasticsearch/test/rest/ESRestTestFeatureService.java [hunk 1]
```diff
@@ -58,10 +58,15 @@ class ESRestTestFeatureService implements TestFeatureService {
...
+        FeatureData featureData = FeatureData.createFromSpecifications(specs);
+        assert featureData.getNodeFeatures().isEmpty()
+            : Strings.format(
+                "Only historical features can be injected via ESRestTestCase#additionalTestOnlyHistoricalFeatures(), rejecting %s",
+                featureData.getNodeFeatures().keySet()
+            );
+        this.knownHistoricalFeatureNames = featureData.getHistoricalFeatures().lastEntry().getValue();
         this.version = nodeVersions.stream().min(Comparator.naturalOrder()).orElse(Version.CURRENT);
-        this.allSupportedFeatures = Sets.union(clusterStateFeatures, historicalFeatures.floorEntry(version).getValue());
+        this.allSupportedFeatures = Sets.union(clusterStateFeatures, featureData.getHistoricalFeatures().floorEntry(version).getValue());
```
Comment (thecoop) [inline test/framework/src/main/java/org/elasticsearch/test/rest/ESRestTestFeatureService.java:69]: nit - this is grabbing the historical features twice, which will (probably) generate the map twice.

Comment (mosche) [reply test/framework/src/main/java/org/elasticsearch/test/rest/ESRestTestFeatureService.java:69]: It's actually a getter, so safe this way 👍

**Pull request merged**